feat(sns): support encryption of topics

yamatatsu opened the pull request:
Add Support KMS masterKey on SNS.
It is allowed to set `masterKey` as `Kms::Key` to `TopicProps`.
Fixes #1729

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://dfq49x5zyi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b407c807-14da-4801-b6c8-b3ddc436de5c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

nija-at commented:
@yamatatsu 

> Is it possible for me to pass this test locally without a security token?

You do not need to run `npm run integ` just yet.  

`integ.sns.ts` synthesizes a CloudFormation template and compares it against the checked in template in [integ.sns.expected.json](https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-sns/test/integ.sns.expected.json). The error you're seeing - `Error: Some stacks have changed.` - happens because you've updated the integration test but not the expected JSON.
Scrollback when you get this error and you will see the diff between what is expected and what is synthesized.

yamatatsu commented on packages/@aws-cdk/aws-sns/test/test.sns.ts:
```diff
@@ -67,25 +68,36 @@ export = {
+      test.done();
+    },
+
     'specify both'(test: Test) {
```
OK. I'll revert it.
I just guessed that this test Intended to test all of `TopicProps` properties.
But I also think it is not needed to extend this test case.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://dfq49x5zyi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=704ce007-e7d3-4e4c-98d9-a1668861177e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://dfq49x5zyi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=60c89c87-ee62-496c-9afc-70396b40049b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://dfq49x5zyi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a21200da-f772-4c65-8aaa-d4ceefaa0dac%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://dfq49x5zyi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1868d44c-ad35-4942-9939-eeddc6d11451%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

yamatatsu commented:
@nija-at 
I fixed the commit message. So it is needed to use `push -f`. Is it your expected? If not, I'll re-create PR.
And I fixed for your comments.

I have a question that I should edit `integ.sns.expected.json` manually? I guess that this json is a snapshot and is edited automatically.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://dfq49x5zyi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=da90c555-2f5e-4e71-b6d0-634357bd87af%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://dfq49x5zyi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=71cbe2b6-2937-4945-b5a3-095a74cb50a1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

nija-at commented:
The team is currently heads down for the next couple of days, so it'll be a bit longer before we can get to this. Apologies for the delayed response on your PR.

nija-at commented:
Code looks fine. We're not accepting community contributions into the CDK until next week, so hang tight.

yamatatsu commented:
OK. It was so fun to complete making this PR with you.
ARIGATO!

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://dfq49x5zyi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a0776c2c-37ca-4f0b-a0fd-435d385002d7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://dfq49x5zyi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a02a9b14-a90b-4107-8b82-07141097d9e8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*
